### ISSUE TITLE
None api version after update Netbox to version 3.0.1

### ISSUE THREAD

**Issue by Sivolen** [bb-Ricardo/netbox-sync#88]
Hi, i have a problem after update NetBox to to version 3.0.1
```

a.gridnev@netbox:/opt/scripts/netbox-sync$ ./netbox-sync.py -l DEBUG2
2021-09-03 09:29:23,159 - INFO: Starting NetBox Sync
2021-09-03 09:29:23,159 - DEBUG: Using config file: /opt/scripts/netbox-sync/settings.ini
2021-09-03 09:29:23,161 - DEBUG: Config: netbox.api_token = 02a***
2021-09-03 09:29:23,161 - DEBUG: Config: netbox.host_fqdn = 10.0.2.18
2021-09-03 09:29:23,161 - DEBUG: Config: netbox.port = 443
2021-09-03 09:29:23,161 - DEBUG: Config: netbox.disable_tls = False
2021-09-03 09:29:23,161 - DEBUG: Config: netbox.validate_tls_certs = False
2021-09-03 09:29:23,161 - DEBUG: Config: netbox.prune_enabled = True
2021-09-03 09:29:23,161 - DEBUG: Config: netbox.prune_delay_in_days = 30
2021-09-03 09:29:23,161 - DEBUG: Config: netbox.default_netbox_result_limit = 200
2021-09-03 09:29:23,162 - DEBUG: Config: netbox.timeout = 30
2021-09-03 09:29:23,162 - DEBUG: Config: netbox.max_retry_attempts = 4
2021-09-03 09:29:23,162 - DEBUG: Config: netbox.use_caching = True
2021-09-03 09:29:23,162 - DEBUG: Created new requests Session for NetBox.
2021-09-03 09:29:23,240 - INFO: Successfully connected to NetBox '10.0.2.18'
2021-09-03 09:29:23,240 - DEBUG: Detected NetBox API version: None
Unable to determine NetBox version, HTTP header 'API-Version' missing.
```
Temporarily changed the code 
```
       #api_version = self.get_api_version()
        api_version = "3.0"
```

**Comment by bb-Ricardo** [comment]
This might have something to do with your reverse proxy setup. I had no issues using this instance: https://github.com/netbox-community/netbox-docker

**Comment by Sivolen** [comment]
Hm, i will try to stop proxy and test again

**Comment by Sivolen** [comment]
Hi, i disabled proxy and the problem persists 

```
a.gridnev@netbox:/opt/scripts/netbox-sync$ ./netbox-sync.py
2021-09-06 10:41:28,064 - INFO: Starting NetBox Sync
2021-09-06 10:41:29,553 - INFO: Successfully connected to NetBox '10.0.2.18'
Unable to determine NetBox version, HTTP header 'API-Version' missing.

a.gridnev@netbox:/opt/scripts/netbox-sync$ env | grep -i proxy
a.gridnev@netbox:/opt/scripts/netbox-sync$ 

```

**Comment by Sivolen** [comment]
Hi!
i tested the script on another server and it has the same error 
In both cases, this is the ubuntu server 20.04
Linux aso 5.11.0-25-generic #27~20.04.1-Ubuntu SMP Tue Jul 13 17:41:23 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

```

  a.gridnev@aso:~/Desktop/netbox-sync$ ./netbox-sync.py
2021-09-07 09:05:56,212 - INFO: Starting NetBox Sync
2021-09-07 09:05:56,471 - INFO: Successfully connected to NetBox '10.0.2.18'
Unable to determine NetBox version, HTTP header 'API-Version' missing.
```

**Comment by bb-Ricardo** [comment]
Hi, I meant the reverse proxy in front of your NetBox Installation. Just try a `curl -I https://10.0.2.18/api/` (or http if unencrypted)
I got following response
```
HTTP/1.1 200 OK
Content-Type: application/json
Vary: Accept, Cookie, Origin
Allow: GET, HEAD, OPTIONS
API-Version: 3.0
X-Content-Type-Options: nosniff
Referrer-Policy: same-origin
X-Frame-Options: SAMEORIGIN
Content-Length: 420
Server: Unit/1.23.0
Date: Tue, 07 Sep 2021 11:12:11 GMT
```

There is a `API-Version` header. This one is missing in your setup. So the script can't determine which version of NetBox you're actually running.

**Comment by gstorme** [comment]
It worked fine on 3.0-beta2, but now I've tried it on 3.0.1 and there is indeed no more API-Version in the response

**Comment by gstorme** [comment]
See https://github.com/netbox-community/netbox/issues/7198

**Comment by bb-Ricardo** [comment]
Interesting find. Thank you for looking into it.

I also just could add the header. What do you think?

**Comment by gstorme** [comment]
Yes indeed, I also think it would be best to just add the `Content-Type: application/json` header to all the netbox api requests

**Comment by Sivolen** [comment]
Hi, i checked my reverse proxy configuration and i don't looking the problem, i would happy if you look this config

    location / {
        proxy_pass http://127.0.0.1:8001;
        proxy_set_header X-Forwarded-Host $http_host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-Proto $scheme;

**Comment by Sivolen** [comment]
> Hi, I meant the reverse proxy in front of your NetBox Installation. Just try a `curl -I https://10.0.2.18/api/` (or http if unencrypted)
> I got following response
> 
> ```
> HTTP/1.1 200 OK
> Content-Type: application/json
> Vary: Accept, Cookie, Origin
> Allow: GET, HEAD, OPTIONS
> API-Version: 3.0
> X-Content-Type-Options: nosniff
> Referrer-Policy: same-origin
> X-Frame-Options: SAMEORIGIN
> Content-Length: 420
> Server: Unit/1.23.0
> Date: Tue, 07 Sep 2021 11:12:11 GMT
> ```
> 
> There is a `API-Version` header. This one is missing in your setup. So the script can't determine which version of NetBox you're actually running.


 Oh, sorry,  yes, no api version 
```

root@netbox:/etc/nginx/sites-available# curl -I http://10.0.2.18/api/
HTTP/1.1 301 Moved Permanently
Server: nginx/1.18.0 (Ubuntu)
Date: Wed, 08 Sep 2021 00:36:02 GMT
Content-Type: text/html
Content-Length: 178
Location: https://10.0.2.18/api/
X-Cache: MISS from localhost
X-Cache-Lookup: HIT from localhost:8080
Via: 1.1 localhost (squid/4.15)
Connection: keep-alive
```

**Comment by bb-Ricardo** [comment]
> 
> Oh, sorry, yes, no api version
> 
> ```
> 
> root@netbox:/etc/nginx/sites-available# curl -I http://10.0.2.18/api/
> HTTP/1.1 301 Moved Permanently
> Server: nginx/1.18.0 (Ubuntu)
> Date: Wed, 08 Sep 2021 00:36:02 GMT
> Content-Type: text/html
> Content-Length: 178
> Location: https://10.0.2.18/api/
> X-Cache: MISS from localhost
> X-Cache-Lookup: HIT from localhost:8080
> Via: 1.1 localhost (squid/4.15)
> Connection: keep-alive
> ```

And if you try `curl -I https://10.0.2.18/api/`

**Comment by Sivolen** [comment]
Hi!
yes 

```
a.gridnev@netbox:~$ curl -I 10.0.2.18/api/
HTTP/1.1 301 Moved Permanently
Server: nginx/1.18.0 (Ubuntu)
Date: Wed, 08 Sep 2021 07:02:50 GMT
Content-Type: text/html
Content-Length: 178
Location: https://10.0.2.18/api/
X-Cache: MISS from localhost
X-Cache-Lookup: HIT from localhost:8080
Via: 1.1 localhost (squid/4.15)
Connection: keep-alive
```

**Comment by bb-Ricardo** [comment]
did you copy the command?

**Comment by Sivolen** [comment]
yes, there's this is how it looks if you make a request for the domain

```
root@netbox:/etc/nginx/sites-available# curl -I https://netbox.aso.rt.local/api/
HTTP/1.1 503 Service Unavailable
Server: squid/4.15
Mime-Version: 1.0
Date: Wed, 08 Sep 2021 22:12:33 GMT
Content-Type: text/html;charset=utf-8
Content-Length: 3739
X-Squid-Error: ERR_DNS_FAIL 0
Vary: Accept-Language
Content-Language: en
```

**Comment by n3bojs4** [comment]
Hello,
Is there any commit to fix this issue ?

Regards.

**Comment by n3bojs4** [comment]
i opened a PR : https://github.com/bb-Ricardo/netbox-sync/pull/93

**Comment by bb-Ricardo** [comment]
Hi, can you try branch `fix/netbox-request-headers`

**Comment by gstorme** [comment]
👍  thanks
`DEBUG: Detected NetBox API version: 3.0`

**Comment by bb-Ricardo** [comment]
fixed in version v1.1.2!
